geopandas/geoseries.py

```diff
@@ -594,7 +594,30 @@ def _constructor(self):
     def _constructor_expanddim(self):
         from geopandas import GeoDataFrame
 
-        return GeoDataFrame
+        def expanddim(data=None, index=None, crs=None, **kwargs):
```

Comment:
Can you maybe define this as a stand-alone function outside of the class definition? (like `_geoseries_constructor_with_fallback`, and so not as a "closure"). I personally like that more, although I can't really articulate why :)

Reply:
I was actually thinking about doing this with the other closure like this in my GeoDataFrame PR. Extra nesting is nice to avoid I guess. I suppose it might be more efficient too? Naively expanddim looks like it would get redefined on every call to `_constructor_expanddim`, but I don't know if python is smart enough to avoid that.

Even if it doesn't matter here, it's probably worth me looking into the asv benchmarks with this set of PRs to see if anything has changed badly.

Reply:
>  I don't know if python is smart enough to avoid that.

Yeah, I also don't fully know how python code execution handles that (whether it is redefined every time, or only the first time when defining/importing the class)